### ISSUE TITLE
Allow download plugins on demand at runtime

### ISSUE THREAD

**Issue by xiangfu0** [apache/pinot#12564]
Current Pinot distribution is getting too large, as well as the docker image size.

We need to figure out a way to just publish shaded plugins and allow download them at the runtime.

This is also good when users just need parquent they don't have to load orc/avro plugins.

**Comment by hpvd** [comment]
yep, another reason for "going small" is security. The less unneeded code is included (own and from dependencies), the fewer security vulnerabilities are delivered or can arise later...
